### PR TITLE
User Connections Page

### LINKED ISSUE #44: User Connections Page
A page containing a list of organizations to which the user is connected and/or is following. The user should be able to unfollow organizations from within the list.

This is what the equivalent page looks like in Gen 2:

image

### PR DESCRIPTION
**Pull request by kristofferlarberg**
This PR contains fetching and rendering of organizations followed by user on `MyOrgsPage`. This is also applied in `MainOrgLayout` where the follow/unfollow button is activated. All fetch functionality for `following` is available via the `useUserFollowing` hook.

Skärmavbild 2021-05-11 kl  14 38 39

Skärmavbild 2021-05-11 kl  14 47 50

Fixes #44

### CODE REVIEW

**Review by richardolsson — CHANGES_REQUESTED** [review]
This looks really good! I like the `useUserFollowing()` hook solution.

Because it's looking so good, I got a little bit picky in my review, so I have a bunch of comments below even though in general all is looking great! ⭐

File: src/pages/my/orgs.tsx
Comment (kristofferlarberg) [reply src/pages/my/orgs.tsx:106]: True!

File: src/fetching/putUserFollowing.ts
Comment (kristofferlarberg) [reply src/fetching/putUserFollowing.ts:19]: Would this mean there is an automatic join request if a user tries to follow without being a member? If so, should there not be some sort of confirmation regarding this before the request is made?

Comment (richardolsson) [reply src/fetching/putUserFollowing.ts:19]: Do you mean user confirmation, i.e. asking the user if they want to connect? Maybe you're right!

File: src/fetching/putUserFollowing.ts
Comment (richardolsson) [reply src/fetching/putUserFollowing.ts:10]: I was suggesting that if it fails, you send a join request instead. But as discussed above, lets hold off on that until #85.

File: src/components/layout/OrgHeader.spec.tsx
Comment (kristofferlarberg) [reply src/components/layout/OrgHeader.spec.tsx:83]: Yes I agree, I'll hide it for now and make a TODO note in connection to it.

File: src/components/UserFollowingList.tsx
Comment (richardolsson) [inline src/components/UserFollowingList.tsx:47]: The "machine-readable" strings `admin` and `organizer` are not meant for humans. Please use human-readable and internationalized versions instead, e.g. using `<Msg id={ `pages.myOrgs.roles.${follow.role}` }/>`.

File: src/components/UserFollowingList.tsx
Comment (richardolsson) [inline src/components/UserFollowingList.tsx:28]: I don't think this is doing what you were hoping to do when you wrote it. This means all organizations have the same key, the literal string _follow.profile.id_.

File: src/components/layout/OrgHeader.tsx
Comment (richardolsson) [inline src/components/layout/OrgHeader.tsx:69]: I'm not used to seeing this approach to rendering an empty element. I think you should just use `null` instead.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Great progress! I found three things that should be fixed before merge. Have a look below. 😊

File: src/components/UserFollowingList.tsx
Comment (richardolsson) [inline src/components/UserFollowingList.tsx:28]: The mistake here wasn't the content of the string, but the fact that it's a constant string literal, i.e. the same exact string key will be used for every organization, not the ID. I'm actually surprised this even renders more than one organization, and doesn't throw warnings in the console (or maybe it does?).

File: src/components/UserFollowingList.tsx
Comment (richardolsson) [inline src/components/UserFollowingList.tsx:49]: I think it would be better to group the role strings together under a common key, the way that I had done in my example in my previous comment, i.e. not just `pages.myOrgs.admin`, but `pages.myOrgs.roles.admin`, etc.

**Review by richardolsson — CHANGES_REQUESTED** [review]
I'm afraid two of the changes you made in your last batch of commits did not fully match what I had requested in my comments. Perhaps I wasn't clear enough, so I've tried to elaborate a little bit here. Please have a look, and let me know if it's not clear. I'd be happy to explain further! 😊 💬

File: src/components/UserFollowingList.tsx
Comment (kristofferlarberg) [reply src/components/UserFollowingList.tsx:28]: Okay! I'm not sure I follow though. `follow.profile.id ` is different on each object, three different organizations are rendered and I don't get any warnings in the console.

File: src/components/UserFollowingList.tsx
Comment (kristofferlarberg) [reply src/components/UserFollowingList.tsx:49]: Of course, I'll change that!

File: src/components/UserFollowingList.tsx
Comment (richardolsson) [reply src/components/UserFollowingList.tsx:28]: I verified that I do indeed get a warning in the console. See below. You should be getting the same warning, in the browser when the component is rendered there, or in the server log when the component is rendered on the server.

image

The javascript expression `follow.organization.id` evaluates the `id` attribute of the object stored in the `organization` attribute of the object called `follow`. You're right that this would be different for every iteration in the loop. The javascript expression `"follow.organization.id"` however is a constant string literal, and will not change.

Comment (kristofferlarberg) [reply src/components/UserFollowingList.tsx:28]: Oh! I see what you mean now 🤦‍♂️ Also seems like warnings was turned of in my console. Sorry about this!

**Review by richardolsson — APPROVED** [review]
This looks good to me now! ✨ 

Great work carrying this all the way across the finish line. 🏃

**Pull request merged**
